pandas/io/parsers.py

```diff
@@ -1296,15 +1296,28 @@ def _validate_usecols_arg(usecols):
     if usecols is not None:
         if callable(usecols):
             return usecols, None
```

Comment:
Given this just returns here is the behavior different by chance when providing a callable argument?

Reply:
So extending the test case you added appending this to the parametrization still failed locally for me on Py2:

```python
 lambda x: x in {u"あああ", u"いい"}
```

Reply:
@WillAyd : The responsibility falls on the user to provide the correct `lambda`.  I'm not really sure what accommodation we can provide besides some painful introspection.